### PR TITLE
OGM-1433 Support for drop collection in native query

### PR DESCRIPTION
**Pull request by raghav-jha**
@DavideD : Please review.

### CODE REVIEW

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBSessionCLIQueryTest.java [hunk 1]
```diff
@@ -1245,4 +1248,46 @@ public void testMapReduceWithOptions() throws Exception {
...
+	@TestForIssue(jiraKey = "OGM-1433")
+	public void testDropCollection() {
+		try ( OgmSession session = openSession() ) {
+			Transaction transaction = session.beginTransaction();
+			final MarkTwainPoem genius = new MarkTwainPoem( 1L, "Genius", "Mark Twain" );
+			session.persist( genius );
+			session.flush();
+			String nativeQuery = "db." + MarkTwainPoem.TABLE_NAME + ".drop()";
+			int result = session.createNativeQuery( nativeQuery ).executeUpdate();
+			assertThat( result ).isEqualTo( 1 );
```
Comment (DavideD) [inline mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBSessionCLIQueryTest.java]: Can you also add a check to make sure that the collection doesn't exist anymore on the db? 
Obtaining the list of collections should look something like this: 
```
		MongoDBDatastoreProvider provider = MongoDBTestHelper.getProvider( sessionFactory );
		MongoIterable<String> listCollectionNames = provider.getDatabase().listCollectionNames();
```

**Comment by DavideD** [comment]
Sorry, it took me so long to check this. Looks really good.
I just have a remark on the test, if you can update it I think we can merge it

**Comment by DavideD** [comment]
If it's OK for you and you are too busy, I can apply those changes and merge it so that it is going to be included in the next release. Let me know. Thanks again

File: mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBSessionCLIQueryTest.java (hunk 1 above)
Comment (raghav-jha) [reply mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBSessionCLIQueryTest.java]: Taken Care.

Comment (DavideD) [reply mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBSessionCLIQueryTest.java]: Cool. thanks

**Comment by raghav-jha** [comment]
@DavideD : Addressed PR comments. Please check.

**Comment by DavideD** [comment]
I've created a new PR with additional changes on top of this one.

Thanks

**Pull request closed without merging**